### DISCUSSION TITLE
/odom topic not working in simulation

### DISCUSSION THREAD

**Discussion by alicia4550** [iRobotEducation/create3_docs#344] [Questions] [answered]
Hi, I have been having trouble using the ```/odom``` topic in the ```create3_sim``` simulation. 

I am using the ```rclpy``` library to create nodes and creating a subscription in my node to subscribe to the ```odom``` topic. Subscribing to the topic works fine when I run my code with the robot, however, when I try to use it when running the simulation, the callback for the subscription is never called. Thus, when I spin the node, nothing happens. No errors have appeared, so I'm unsure where to go from here.

I have tried subscribing to other topics, using publishers, using actions, etc. in the simulation, all of which work as expected, so it seems like it is only the ```/odom``` topic that does not work.

Any ideas on what the problem is and how to resolve this (or if more information is needed)? Thanks.

I'm using Humble and running on Ubuntu 22.04.

**Comment by justinIRBT** [comment, marked as answer]
I wonder if your issue is tied to what this simulator PR is fixing:
https://github.com/iRobotEducation/create3_sim/pull/206

**Reply by justinIRBT** [reply]
Note, this was merged if you want to try again.
